src/scripts/audit/tool.sh

```diff
@@ -0,0 +1,305 @@
+    validationError "Unable to locate the image file ${imageFile} specified with '-i' parameter"
+  fi
+
+  if [[ -n "${kubernetesFile}" && ! -f ${kubernetesFile} ]]; then
+    validationError "Unable to locate the Kubernetes Node information file ${kubernetesFile} specified with '-k' parameter."
+  fi
+
+  if [ ! -f "$artifactsFile" ]; then
+    validationError "Unable to locate the artifacts file ${artifactsFile}"
+  fi
+
+  failIfValidationErrors
+
+  # Initialize workspace dir and generate report header
+  initWorkDir
+  generateReportHeader
+
+  # If image list is not provided, get all images on the current machine 
+  if [ -z "$imageFile" ]; then
+    echo "Scanning all images on current machine, use '-i' option if you want to scan specific set of images."
+    imageFile=$(docker images --digests --format "{{.Repository}}:{{.Tag}}@{{.Digest}}@{{.ID}}" | grep -v "<none>:<none>")
+  fi
+
+  # Validate list of available images
+  if [[ -n "${kubernetesFile}" && ${disableImageValidation} == 'false' ]]; then
+    validateAvailableImages
+  fi
+
+  # Read or create image file
+  if [ -z "$imageFile" ]; then
+    IFS=$'\n' read -d '' -r -a images < "${imageFile}"
+    imageFile=${images[@]}
+  fi
+
+  # Read artifacts file
+  IFS=$'\n' read -d '' -r -a artifacts < "$artifactsFile"
+  artifactsFile=${artifacts[@]}
+
+  # Read image to node mapping from kubernetes file in nodeToImageMapping array
+  if [ -n "${kubernetesFile}" ]; then
+    getImagesInNodeFile
+  fi
+
+}
+
+function validateAvailableImages {
+
+  info "Validating images available for scanning on this machine..."
+  getMissingImagesList
+  missingImageCount=$?
+  if [[ $missingImageCount -gt 0 ]]; then
+    fail "Found $missingImageCount images missing on current machine. The list of missing images is saved in ${missingImagesReport} file, these images must be pulled to current machine before the script can continue. Use '-d' option to disable image validation."
+  fi
+}
+
+#
+# Initialize workspace dir and generate report header
+#
+function initWorkDir {
+  if [ -z ${workDir} ]; then
+    workDir="./work"
+  fi
+  rm -fr "${workDir}"
+  mkdir -p "${workDir}"
+  missingImagesReport="${workDir}/missing-images.txt"
+  reportFile="${workDir}/report.txt"
+  logFile=${workDir}/tool.log
+  echo "Log for this run can be found in ${logFile}"
+  info "Image scan started at `date -u '+%Y-%m-%d %H.%M.%SUTC'`"
+}
+
+#
+# Read image list, verify hash and get node names (if node info provided) and process
+#
+function scanAndSearchImageList {
+
+  for image in ${imageFile}
+  do
+    imageRepoTag=$(echo $image | cut -d'@' -f1)
+    imageHash=$(echo $image | cut -d'@' -f2)
+    imageId=$(echo $image | cut -d'@' -f3)
+    repository=$(echo $imageRepoTag | cut -d':' -f1)
+    tag=$(echo $imageRepoTag | cut -d':' -f2)
+
+    if [ -z ${imageId} ]; then
+      imageId=$(getImageId "${imageRepoTag}")
+    fi
+
+    imageSize=`docker inspect ${imageId} | jq .[].Size`
```

Comment:
I should have mentioned this earlier, but we will need to support other command-line tools, such as podman.

Reply:
I tried running `podman images` command in Openshift cluster to understand it's usage. It required me to run as `sudo` and it doesn't return any images when run as a regular user. I'm not sure about other environment. Do you know if there's any other environment where I can test `podman`?

Reply:
I have incorporated changes to support other command-line tools such as podman. The tool can be run with "-m podman" option to replace docker with podman. I have tested tool with 'podman' on Openshift helper node.